From ba41fceedc94b038fb35f17b9e41e6677c3bce88 Mon Sep 17 00:00:00 2001
From: JD <100024149@ge.com>
Date: Fri, 12 May 2017 13:13:50 +0200
Subject: [PATCH] Add install target for buildroot

---
 RPi_utils/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/RPi_utils/Makefile b/RPi_utils/Makefile
index 95ceb2f..e0d4e0a 100644
--- a/RPi_utils/Makefile
+++ b/RPi_utils/Makefile
@@ -17,3 +17,5 @@ RFSniffer: ../rc-switch/RCSwitch.o RFSniffer.o
 clean:
 	$(RM) ../rc-switch/*.o *.o send codesend servo RFSniffer
 
+install:
+	cp send codesend RFSniffer /usr/bin/
-- 
2.10.2

